state-chain/runtime/src/chainflip.rs

```diff
@@ -315,21 +315,29 @@ impl TransactionBuilder<Bitcoin, BitcoinApi<BtcEnvironment>> for BtcTransactionB
 pub struct SolanaTransactionBuilder;
 impl TransactionBuilder<Solana, SolanaApi<SolEnvironment>> for SolanaTransactionBuilder {
 	fn build_transaction(
-		_signed_call: &SolanaApi<SolEnvironment>,
+		signed_call: &SolanaApi<SolEnvironment>,
 	) -> <Solana as Chain>::Transaction {
-		todo!()
+		signed_call.transaction.clone()
 	}
+
 	fn refresh_unsigned_data(_tx: &mut <Solana as Chain>::Transaction) {
-		todo!()
+		// Transaction fee on SOlana is a fixed 5k lamports per signature. Since we always have
+		// a single signature, therefore the transaction fee does not change.
```

Comment:
I'm not sure about how this works. From the name it sounds like it's all the data before signing, but there is this comment in the code: `Note that calldata cannot be updated, or it would invalidate the signature.`
For Solana, if this is before any signature is done, then we should refresh the priority fee (as you say the base fee doesn't need any update).
However, if the comment is correct and this is after we have a signature, then we shouldn't change the priority fee, as that would require a new signature.
Do you know which one is it?

Reply:
This is currently only used by EVM chains where gas fee and priority fee is updated.

To answer your question - once the transaction content is changed (fees updated), the next function: `requires_signature_refresh` will return true, and we will then re-sign the call.

This refresh fn is called just before when a call is broadcasted(in the `fn start_broadcast_attempt()` - so it ensures the call always have the correct gas fee setting. If the gas fee changed, the call is re-signed will be tried again.

Reply:
So the question here would be - how fast do fees change in Solana and how accurate do we have to be? Basically we are taking the lesser evil here:
1. risk of inaccurate gas setting causing the transaction to fail 
2. cost of resigning a call

Reply:
Gotcha, thanks for the explanation.
Then we shouldn't use this for Solana but not for the reason explained in the comment. We can't use it because refreshing the priority fee would require resigning, which we don't want to do.